Content/NPCs/Misc/LootWraith.cs

```diff
@@ -0,0 +1,436 @@
+			chain.UpdateChain();
+		}
+
+		private void ManageCache()
+		{
+			cache = new List<Vector2>();
+
+			float pointLength = TotalLength(GetChainPoints()) / NUM_SEGMENTS;
+
+			float pointCounter = 0;
+
+			int presision = 30; //This normalizes length between points so it doesnt squash super weirdly on certain parts
+			int iMax = NUM_SEGMENTS - 1;
+			if (enraged)
+				iMax--;
+			for (int i = 0; i < iMax; i++)
+			{
+				for (int j = 0; j < presision; j++)
+				{
+					pointCounter += (chain.ropeSegments[i].posNow - chain.ropeSegments[i + 1].posNow).Length() / presision;
+
+					while (pointCounter > pointLength)
+					{
+						float lerper = j / (float)presision;
+						cache.Add(Vector2.Lerp(chain.ropeSegments[i].posNow, chain.ropeSegments[i + 1].posNow, lerper));
+						pointCounter -= pointLength;
+					}
+				}
+			}
+
+			while (cache.Count < NUM_SEGMENTS)
+			{
+				cache.Add(chain.ropeSegments[iMax].posNow);
+			}
+
+			while (cache.Count > NUM_SEGMENTS)
+			{
+				cache.RemoveAt(cache.Count - 1);
+			}
+		}
+
+		private void ManageTrail()
+		{
+			trail ??= new Trail(Main.instance.GraphicsDevice, NUM_SEGMENTS, new TriangularTip(1), factor => 7, factor => Lighting.GetColor((int)(NPC.Center.X / 16), (int)(NPC.Center.Y / 16)) * MathF.Sqrt(1 - factor.X));
+
+			List<Vector2> positions = cache;
+			trail.NextPosition = NPC.Center;
+
+			trail.Positions = positions.ToArray();
+
+			trail2 ??= new Trail(Main.instance.GraphicsDevice, NUM_SEGMENTS, new TriangularTip(1), factor => 7, factor => Color.White * chargeupCounter * MathF.Sqrt(1 - factor.X));
+
+			trail2.NextPosition = NPC.Center;
+
+			trail2.Positions = positions.ToArray();
+		}
+
+		private void DrawChain()
+		{
+			if (trail == null || trail == default)
+				return;
+
+			Main.spriteBatch.End();
+			Effect effect = Terraria.Graphics.Effects.Filters.Scene["RepeatingChain"].GetShader().Shader;
+
+			var world = Matrix.CreateTranslation(-Main.screenPosition.Vec3());
+			Matrix view = Main.GameViewMatrix.ZoomMatrix;
+			var projection = Matrix.CreateOrthographicOffCenter(0, Main.screenWidth, Main.screenHeight, 0, -1, 1);
+
+			effect.Parameters["transformMatrix"].SetValue(world * view * projection);
+			effect.Parameters["sampleTexture"].SetValue(ModContent.Request<Texture2D>(Texture + "_Chain").Value);
+			effect.Parameters["flip"].SetValue(false);
+			effect.Parameters["alpha"].SetValue(1);
+
+			List<Vector2> points;
+
+			if (cache == null)
+				points = GetChainPoints();
+			else
+				points = trail.Positions.ToList();
+
+			effect.Parameters["repeats"].SetValue(TotalLength(points) / 20f);
+
+			BlendState oldState = Main.graphics.GraphicsDevice.BlendState;
+			Main.graphics.GraphicsDevice.BlendState = BlendState.Additive;
+			trail?.Render(effect);
+			Main.graphics.GraphicsDevice.BlendState = oldState;
+
+			effect.Parameters["sampleTexture"].SetValue(ModContent.Request<Texture2D>(Texture + "_Chain_White").Value);
+
+			//trail2?.Render(effect);
```

Comment:
if trail2 is never rendered, why are we updating it? Is this fully unused? If so, the update code for it should be removed.